modules/custom/az_migration/src/EventSubscriber/MigrationSubscriber.php

```diff
@@ -0,0 +1,141 @@
+use Drupal\Core\Entity\EntityFieldManager;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\migrate\Event\MigrateRollbackEvent;
+
+/**
+ * Class MigrationSubscriber.
+ *
+ * Handles various migrations tasks outside of normal flow.
+ *
+ * @package Drupal\az_migration
+ */
+class MigrationSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The entity manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The entity field manager.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManager
+   */
+  protected $entityFieldManager;
+
+  /**
+   * The entity type.
+   *
+   * @var string
+   */
+  protected $entityType;
+
+  /**
+   * The entity bundle.
+   *
+   * @var string
+   */
+  protected $bundle;
+
+  /**
+   * Constructs a new MigrationSubscriber.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity manager.
+   * @param \Drupal\Core\Entity\EntityFieldManager $entity_field_manager
+   *   The entity field manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, EntityFieldManager $entity_field_manager) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->entityFieldManager = $entity_field_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    $events[MigrateEvents::PRE_ROLLBACK][] = ['onMigratePreRollback'];
+    return $events;
+  }
+
+  /**
+   * React to rollback start.
+   *
+   * @param \Drupal\migrate\Event\MigrateRollbackEvent $event
+   *   The map event.
+   */
+  public function onMigratePreRollback(MigrateRollbackEvent $event) {
+    $dest = $event->getMigration()->getDestinationConfiguration();
+    if (!isset($dest['plugin']) && !isset($dest['bundle'])) {
+      return;
+    }
+    // Grab our type and make the magiz happen.
+    $this->entityType = ltrim(strstr($dest['plugin'], ':'), ':');
+    $this->bundle = $dest['bundle'];
+    $this->checkFieldsforMediaEntities();
+  }
+
+  /**
+   * Checks the nodes fields for media entities.
+   */
+  private function checkFieldsforMediaEntities() {
+    // Grab all our fields for this entity type.
+    $fields = $this->entityFieldManager
+      ->getFieldDefinitions($this->entityType, $this->bundle);
+
+    foreach ($fields as $field_name => $field_definition) {
+      /** @var \Drupal\field\Entity\FieldConfig $field_definition */
+      if ($field_definition->getTargetBundle() !== NULL) {
+        if ($field_definition->getType() === 'entity_reference'
+            && $field_definition->getFieldStorageDefinition()->getSetting('target_type') === 'media') {
+          $this->removeMediaEntities($field_name);
+        }
+      }
+    }
+  }
+
+  /**
+   * Remove the media entities for that field and type.
+   *
+   * @param string $field_name
+   *   The field name we are checking.
+   */
+  private function removeMediaEntities($field_name) {
```

Comment:
There should be cases where other fields or content types should reference the same piece of media content as media can be shared and reused. Trying to manually compute a rollback in an event listener doesn't seem necessary.

I'd recommend migration of media be its own migration that takes place before the person migration; rolling back person doesn't necessarily mean media needs to be deleted. That would let the media migration keep track of which content came from the source system via its own migration map table.

Reply:
Tried to fetch the wanted media as per the content, using the existing file entities. So It is fully dependent on content migration mapping, that is why removing the media in rollback. Using the existing file entity won't duplicate the files, and also here not deleting the files in this rollback. Thanks.